src/qtils/strict_sptr.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <stdexcept>
+#include <type_traits>
+
+namespace qtils {
+
+  /**
+   * @brief A strict version of std::shared_ptr that guarantees non-null values.
+   *
+   * This class prevents default construction and ensures that the shared
+   * pointer is never null. It removes unsafe methods such as `reset()` and
+   * `operator=`. Provides safe access to the underlying std::shared_ptr.
+   *
+   * @tparam T The type of the managed object.
+   */
+  template <class T>
+    requires(not std::is_void_v<T>)
+  class StrictSharedPtr {
+   public:
+    // Default constructor is deleted to prevent an uninitialized state
+    StrictSharedPtr() = delete;
+
+    // Explicitly delete constructor from nullptr
+    StrictSharedPtr(std::nullptr_t) = delete;
+
+    /**
+     * @brief Constructs a StrictSharedPtr from an existing std::shared_ptr.
+     *
+     * @param sptr The shared pointer to be wrapped.
+     * @throws std::invalid_argument if the pointer is null.
+     */
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    StrictSharedPtr(std::shared_ptr<T> sptr) : ptr_(std::move(sptr)) {
+      if (!ptr_) {
+        throw std::invalid_argument("Shared pointer is null");
+      }
+    }
+
+   private:
+    struct ForceInject {};  // Auxiliary arg appended to select ctor for DI
```

Comment:
Please elaborate on it a bit so a person not closely familiar with Boost.DI would understand.